libs/sdk-ui-kit/src/@ui/UiMenu/components/defaults/DefaultUiMenuContent.tsx

```diff
@@ -0,0 +1,42 @@
+// (C) 2025 GoodData Corporation
+
+import React from "react";
+import { e } from "../../menuBem.js";
+import { IUiMenuContentItem, IUiMenuItemData } from "../../types.js";
+import { typedUiMenuContextStore } from "../../context.js";
+import { DefaultUiMenuHeader } from "./DefaultUiMenuHeader.js";
+
+/**
+ * Container for rendering custom content in menu.
+ * @internal
+ */
+export const DefaultUiMenuContent = React.memo(function DefaultUiMenuContent<
+    T extends IUiMenuItemData = object,
+>({ item }: { item: IUiMenuContentItem<T> }): React.ReactElement {
+    const { useContextStore, createSelector } = typedUiMenuContextStore<T>();
+    const selector = createSelector((ctx) => ({
+        onClose: ctx.onClose,
+        setShownCustomContentItemId: ctx.setShownCustomContentItemId,
+        setFocusedId: ctx.setFocusedId,
+        shownCustomContentItemId: ctx.shownCustomContentItemId,
+    }));
+
+    const { onClose, setShownCustomContentItemId, setFocusedId, shownCustomContentItemId } =
+        useContextStore(selector);
+
+    const handleBack = React.useCallback(() => {
+        setFocusedId(shownCustomContentItemId);
+        setShownCustomContentItemId(undefined);
+    }, [setShownCustomContentItemId, shownCustomContentItemId, setFocusedId]);
+
+    const ContentComponent = item.component;
+
+    return (
+        <>
+            {item.showComponentOnly !== true && <DefaultUiMenuHeader />}
```

Comment:
It would be better to do `{item.showComponentOnly ? <DefaultUiMenuHeader /> : null}. This construction with negated check for true and then AND to component does not look good.